### PR TITLE
Allow for generation of a constrained float with multiple_of argument for hypothesis plugin

### PR DESCRIPTION
**Pull request by tobi-lipede-oodle**
## Change Summary

This enables Hypothesis to generate a `ConstrainedFloat` with the `multiple_of` argument enabled.

## Related issue number
https://github.com/samuelcolvin/pydantic/discussions/2443

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

**Comment by samuelcolvin** [comment]
is there any issue for this?

**Comment by tobi-lipede-oodle** [comment]
Sorry! I'll make that now

**Comment by PrettyWood** [comment]
@samuelcolvin I didn't add the `v1.8.1` label since nothing is broken but it would make sense to add it. Being able to use `multiple_of` only with `int` and not `float` with `hypothesis` is weird

**Comment by samuelcolvin** [comment]
may as well include this in v1.8.1

**Comment by samuelcolvin** [comment]
thanks so much.

File: pydantic/_hypothesis_plugin.py [hunk 1]
```diff
@@ -284,7 +285,21 @@ def resolve_confloat(cls):  # type: ignore[no-untyped-def]
...
+
+    if cls.multiple_of is None:
+        return st.floats(min_value, max_value, exclude_min=exclude_min, exclude_max=exclude_max, allow_nan=False)
+
+    if min_value is not None:
+        min_value = math.ceil(min_value / cls.multiple_of)
+        if exclude_min:
+            min_value = min_value + 1
+    if max_value is not None:
+        assert max_value >= cls.multiple_of, 'Cannot build model with max value smaller than multiple of'
```
Comment (Stranger6667) [inline pydantic/_hypothesis_plugin.py:297]: This assertion restricts the generation more than needed. For example, `max_value=-2.0` and `multiple_of=-1.0` will fail this assertion, but if the plugin generates `-3.0`, it satisfies the constraints.

Generally, I suggest taking a look at `multipleOf` [implementation](https://github.com/Zac-HD/hypothesis-jsonschema/blob/master/src/hypothesis_jsonschema/_canonicalise.py#L699) in `hypothesis-jsonschema` (there are multiple places in the linked file) as it handles all these constraints.

Comment (tobi-lipede-oodle) [reply pydantic/_hypothesis_plugin.py:297]: thanks, will take a look

**Pull request merged**